vpr/src/noc/odd_even_routing.cpp

```diff
@@ -1,152 +1,75 @@
 
 #include "odd_even_routing.h"
-#include "globals.h"
+
 #include "move_utils.h"
 
+#include <set>
+#include <map>
+
+OddEvenRouting::OddEvenRouting(const NocStorage& noc_model) {
+    std::set<int> unique_x, unique_y, unique_z;
+
+    for (const NocRouter& router : noc_model.get_noc_routers()) {
+        t_physical_tile_loc loc = router.get_router_physical_location();
+        unique_x.insert(loc.x);
+        unique_y.insert(loc.y);
+        unique_z.insert(loc.layer_num);
+    }
+
```

Comment:
Comment what this does --> creates a map with each entry in the passed in set having a value assigned that is a unique integer starting at 0